--- ./WEB-INF/jsp/modules/viewReport/DefaultJasperViewerState.jsp	2012-05-28 07:33:18.000000000 +0530
+++ ./WEB-INF/jsp/modules/viewReport/DefaultJasperViewerState_WithExportPermissions.jsp	2013-02-04 11:07:54.485185468 +0530
@@ -25,6 +25,7 @@
 --%>
 
 <%@ taglib prefix="c" uri="http://java.sun.com/jstl/core_rt" %>
+<%@ taglib prefix="authz" uri="http://www.springframework.org/security/tags" %>
 
 <%@ page import="net.sf.jasperreports.engine.export.*" %>
 <%@ page import="net.sf.jasperreports.engine.*" %>
@@ -65,7 +66,9 @@
   }
 
   script.append("Report.exportersList = [];");
-  if (!isEmptyReport) {
+  %>
+  	<authz:authorize ifAnyGranted="ROLE_ADMINISTRATOR,ROLE_ALLOW_EXPORT">
+  <% if (!isEmptyReport) {
      Map configuredExporters = (Map) request.getAttribute("configuredExporters");
      for (Object configuredExporterEntry: configuredExporters.entrySet()) {
         Map.Entry configuredExporter = (Map.Entry) configuredExporterEntry;
@@ -98,7 +101,9 @@
         script.append("]}');");
      }
   }
-
+  %>
+  	</authz:authorize>
+  <%
   pageContext.setAttribute("evalScript", script.toString());
 %>
 
